a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -264,16 +264,22 @@ class ConfigureVRRPSync(VThunderBaseTask):
     @axapi_client_decorator
     def execute(self, vthunder, backup_vthunder):
         """Execute to sync up vrrp in two vThunder devices."""
-        try:
-            self.axapi_client.system.action.configSynch(backup_vthunder.ip_address,
-                                                        backup_vthunder.username,
-                                                        backup_vthunder.password)
-            LOG.debug("Waiting 30 seconds for config synch.")
-            time.sleep(30)
-            LOG.debug("Sync up for vThunder master")
-        except (acos_errors.ACOSException, req_exceptions.ConnectionError) as e:
-            LOG.exception("Failed VRRP sync: %s", str(e))
-            raise e
+        attempts = CONF.a10_controller_worker.amp_active_retries
+        while attempts >= 0:
+            try:
+                attempts = attempts - 1
+                self.axapi_client.system.action.configSynch(backup_vthunder.ip_address,
+                                                            backup_vthunder.username,
+                                                            backup_vthunder.password)
+                LOG.debug("Waiting 30 seconds for config synch.")
+                time.sleep(30)
+                LOG.debug("Sync up for vThunder master")
+                break
+            except (acos_errors.ACOSException, req_exceptions.ConnectionError) as e:
+                if attempts < 0:
+                    LOG.exception("Failed VRRP sync: %s", str(e))
+                    raise e
+                time.sleep(CONF.a10_controller_worker.amp_active_wait_sec)
```

Comment:
no sleep is needed. As we can see in the log below, after we issue configuration sync at "May 05 07:48:00" vthunder tried to sync. configuration until "May 05 07:51:14". So, vthunder itself has some timeout mechanism.

```
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]: DEBUG acos_client.v30.axapi_http [-] axapi_http: full url = https://192.168.0.165:443/axapi/v3/configure/sync {{(pid=7702) request_impl build/bdist.linux-x86_64/egg/acos_client/v30/ax
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]: DEBUG acos_client.v30.axapi_http [-] axapi_http: POST url = /axapi/v3/configure/sync {{(pid=7702) request_impl build/bdist.linux-x86_64/egg/acos_client/v30/axapi_http.py:76}}
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]: DEBUG acos_client.v30.axapi_http [-] axapi_http: params = {
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:     "sync": {
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "pwd": "a10",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "address": "192.168.0.78",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "type": "all",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "usr": "admin",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "auto-authentication": 1
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:     }
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]: } {{(pid=7702) request_impl build/bdist.linux-x86_64/egg/acos_client/v30/axapi_http.py:77}}
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]: DEBUG acos_client.v30.axapi_http [-] axapi_http: params + axapi_args = {
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:     "sync": {
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "pwd": "a10",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "address": "192.168.0.78",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "type": "all",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "usr": "admin",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:         "auto-authentication": 1
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:     }
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]: } {{(pid=7702) request_impl build/bdist.linux-x86_64/egg/acos_client/v30/axapi_http.py:86}}
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]: DEBUG acos_client.v30.axapi_http [-] axapi_http: params_all = {u'sync': {u'pwd': u'a10', u'address': u'192.168.0.78', u'type': u'all', u'usr': u'admin', u'auto-authentication': 1}} {{
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]: DEBUG acos_client.v30.axapi_http [-] axapi_http: headers = {
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:     "Content-type": "application/json",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:     "Authorization": "A10 61ff168286875f528f3230cc92a1ef",
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]:     "User-Agent": "ACOS-Client-AGENT-2.6.1"
May 05 07:48:00 openstack-3 a10-octavia-worker[7659]: } {{(pid=7702) request_impl build/bdist.linux-x86_64/egg/acos_client/v30/axapi_http.py:108}}
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]: DEBUG acos_client.v30.axapi_http [-] axapi_http: data = {
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]:     "response": {
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]:         "status": "fail",
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]:         "err": {
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]:             "msg": "Configuration sync failed",
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]:             "code": 672268402,
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]:             "from": "BACKEND"
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]:         }
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]:     }
May 05 07:51:14 openstack-3 a10-octavia-worker[7659]: } {{(pid=7702) request_impl build/bdist.linux-x86_64/egg/acos_client/v30/axapi_http.py:146}}

```